articlequality/extractors/ukwiki.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import re
+import traceback
+
+from .extractor import TemplateExtractor
+
+logger = logging.getLogger(__name__)
+
+NORMALIZED_LABELS = {
+        "ВС": ["ВС", "вс", "Вибрана стаття", "вибрана стаття"],
+        "ДС": ["ДС", "дс", "Добра стаття", "добра стаття"],
+        "I": ["I", "1"],
+        "II": ["II", "2"],
+        "III": ["III", "3"],
+        "IV": ["IV", "4", "Stub", "stub"]
+}
+
+
+def from_template(template):
+
+    if template.has_param('class') or template.has_param('рівень'):
+        template_name = normalize_template_name(template.name)
+        project_name = normalize_project_name(template_name)
+        try:
+            label = get_quality(template)
+
+            if label in POSSIBLE_LABELS:
+                yield (project_name, label)
+            else:
+                logger.debug("Class {0} not in possible classes."
+                              .format(label))
+                pass  # not a quality assessment class
+
+        except ValueError as e:
+            logger.warning(traceback.format_exc())
+            pass  # no assessment class in template
+
+
+def get_quality(template):
+    LABEL_MAP = {observed_label: normalized_label
+             for normalized_label, observed_labels in 
+             NORMALIZED_LABELS.items()
+             for observed_label in observed_labels}
```

Comment:
You want this LABEL_MAP at the top so that it is only generated once.  Then you'll use the LABEL_MAP here to check to see if a label from a "class" or "рівень" matches an expected string (keys of LABEL_MAP) and to convert it to the canonical form (values of LABEL_MAP).

Reply:
```
if template.has_param("рівень") or template.has_param("class"):
  if template.has_param("class"): 
    label = template.get_param("рівень")
  else:
    label = template.has_param("class")
  return LABEL_MAP.get(label)
else:
  return None
```